orchestrator/README.md

```diff
@@ -79,3 +80,22 @@ be omitted if the default behavior is desired.
 The orchestrator runs as a non-root user. As such, the container must be run with permissions of the `docker` group on the host
 in order to access the socket. The socket must also be mounted into the container (the bind mount `/var/run/docker.sock:/var/run/docker.sock:ro`).
 This docker group ID changes from system to system and needs to be checked manually using a command like `cat /etc/group | grep docker | awk -F: '{ print $3 }'`
+
+## Cache Implementation
+
+IMAGE uses Memcached as in-memory data store. Cache is implemented using [MemJS](https://www.npmjs.com/package/memjs). Following is the confugration to enable Cache for preprocessors:
+
+- Cache size is configured in the docker-compose in the commad attribute under memcached service `command: -m 4096` implies cache size of 4GB.
+
+- Cache timeout is configured at the preprocessor level, with the label `ca.mcgill.a11y.image.cacheTimeout` . Label value is the timeout value in seconds. Timeout value of 0 indicates that Cache is disabled for a preprocessor. Missing `ca.mcgill.a11y.image.cacheTimeout` label on the preprocessor will default to timeout value of 0.
+
+- Cache key is generated using the following attributes:
+  - `reqData` can have the following falues
+    - `request["data"]` (for graphics)
+    - `request["placeID"]`/`request["coordinates"]` (for maps)
+    - `request["highChartsData"]` (for charts)
+  - `preprocessor` - preprocessor id as returned in the response
```

Comment:
For clarity, this ID is **not** identifying the preprocessor itself, it is identifying the _data returned by the preprocessor_. This is why we have greated multiple object detection preprocessors that all use the same ID, they are separate components that produce data with the same semantic meaning and the same formatting.